src/main/kotlin/org/malachite/estella/services/TaskService.kt

```diff
@@ -62,19 +80,31 @@ class TaskService(
         taskRepository
             .findById(taskId)
             .get()
-            .tests.binaryStream
-            .readAllBytes()
-            .contentToString()
-            .let { Json.decodeFromString<List<TaskTestCaseDto>>(it) }
+            .tests
+            .toBase64String()
+            .let { Json.decodeFromString(it) }
 
-    fun addTask(task: TaskDto) =
+    fun addTask(organizationUuid: UUID, taskDto: TaskDto) =
         withExceptionThrower {
-            if (!checkTestsFormat(task.testsBase64)) throw IllegalArgumentException("Tests in wrong format")
-            taskRepository.save(task.toTask())
+            // Task saving
```

Comment:
Ogólnie rzecz biorąc, jak musisz pisać komentarze, żeby powiedzieć, co dany kawałek metody robi, to my się tu nie bawimy w optymalizację kodu i można je po prostu do metody wydzielić.